### PR TITLE
Fix the amount of cookies

### PR DESCRIPTION
**Pull request by fiammybe**
put into place getcookie() and setcookie() to clean up the way the filtering is stored in functional cookies

### CODE REVIEW

File: htdocs/libraries/icms/ipf/view/Table.php [hunk 1]
```diff
@@ -656,20 +676,20 @@ public function render($fetchOnly = false, $debug = false) {
 
 		$this->setSortOrder();
 
-		$this->_limitsel = !empty($limitsel) ? $limitsel : icms_getCookieVar($script_name . '_limitsel', '15');
+		$this->_limitsel = $_GET['limitsel'] ?? $this->getCookie('limitsel', '15');
```
Comment (MekDrop) [inline htdocs/libraries/icms/ipf/view/Table.php:679]: ?? works only in PHP 7.0

File: htdocs/libraries/icms/ipf/view/Table.php [hunk 2]
```diff
@@ -582,6 +570,38 @@ public function renderForPrint() {
...
+	 *
+	 * @param string $fieldName Field name read from cookie
+	 * @param string|null $defaultValue Default value
+	 *
+	 * @return string
+	 */
+	protected function getCookie(string $fieldName, string $defaultValue = null) {
+		$name = 'tbl_' . str_replace('.', '_', $fieldName);
+
+		return $_COOKIE[$name] ?? $defaultValue;
```
Comment (MekDrop) [inline htdocs/libraries/icms/ipf/view/Table.php:584]: ?? works only in PHP 7.0

**Review by MekDrop — CHANGES_REQUESTED** [review]
Overall good but I found are two minor problems that will prevent to run this on PHP 5.6

File: htdocs/libraries/icms/ipf/view/Table.php [hunk 3]
```diff
@@ -204,14 +204,10 @@ public function setDefaultOrder($default_order) {
...
-	 * @todo	change to dependency injection methods
+	 * @return string the default order returned from a cookie
 	 */
-	public function getDefaultOrder() {
-		if ($this->_ordersel) {
-			return icms_getCookieVar($_SERVER['SCRIPT_NAME'] . '_' . $this->_id . '_ordersel', $this->_ordersel);
-		} else {
-			return icms_getCookieVar($_SERVER['SCRIPT_NAME'] . '_' . $this->_id . '_ordersel', 'ASC');
-		}
+	public function getDefaultOrder(): string {
```
Comment (MekDrop) [inline htdocs/libraries/icms/ipf/view/Table.php:209]: Return types are PHP 7.x only

File: htdocs/libraries/icms/ipf/view/Table.php (hunk 1 above)
Comment (fiammybe) [reply htdocs/libraries/icms/ipf/view/Table.php]: changed it to isset($_GET['limitsel']) ? $_GET['limitsel'] :  $this->getCookie('limitsel', '15');

File: htdocs/libraries/icms/ipf/view/Table.php (hunk 2 above)
Comment (fiammybe) [reply htdocs/libraries/icms/ipf/view/Table.php]: changed it to return isset($_COOKIE[$name]) ? $_COOKIE[$name] : $defaultValue;

File: htdocs/libraries/icms/ipf/view/Table.php (hunk 3 above)
Comment (fiammybe) [reply htdocs/libraries/icms/ipf/view/Table.php]: removed

**Review by MekDrop — APPROVED** [review]

**Comment by MekDrop** [comment]
Fixes #840

**Pull request merged**